model/riscv_insts_vext_vset.sail

```diff
@@ -137,26 +127,28 @@ function clause execute VSETI_TYPE(ma, ta, sew, lmul, uimm, rd) = {
   /* set vtype and calculate VLMAX */
   let vtype_reserved = zeros(sizeof(xlen) - 9);
   let vtype_val : xlenbits = 0b0 @ vtype_reserved @ ma @ ta @ sew @ lmul;
-  writeCSR(csr_name_map("vtype"), vtype_val);
+  writeCSR(0xC21, vtype_val);
 
-  let LMUL_new  : real = get_vtype_LMUL();
-  let SEW_new   : int = get_vtype_vsew();
-  VLMAX = floor(LMUL_new) * vlen / SEW_new; /* VLMAX = LUML * VLEN / SEW */
-  AVL = unsigned(sail_zero_extend(uimm, sizeof(xlen))); /* AVL is encoded as 5-bit zero-extended imm in the rs1 field */
+  let LMUL_new : real = get_vtype_LMUL();
+  let SEW_new  : int = get_vtype_vsew();
+  let VLMAX    : int = floor(LMUL_new) * vlen / SEW_new;
+  let AVL      : int = unsigned(sail_zero_extend(uimm, sizeof(xlen))); /* AVL is encoded as 5-bit zero-extended imm in the rs1 field */
 
   /* set vl according to VLMAX and AVL */
   if (AVL <= VLMAX) then {
-    writeCSR(csr_name_map("vl"), to_bits(sizeof(xlen), AVL));
+    writeCSR(0xC20, to_bits(sizeof(xlen), AVL));
     X(rd) = to_bits(sizeof(xlen), AVL)
   } else if (AVL < 2 * VLMAX) then { /* ceil(AVL / 2) ≤ vl ≤ VLMAX */
-    writeCSR(csr_name_map("vl"), to_bits(sizeof(xlen), ceil(to_real(AVL) / 2.0)));
+    writeCSR(0xC20, to_bits(sizeof(xlen), ceil(to_real(AVL) / 2.0)));
```

Comment:
So (AVL + 1) / 2. Don't use floats like this, they cause all kinds of issues and it's just lazy.

Reply:
Fixed